### PR TITLE
[WIP] Documentation update

### PR DESCRIPTION
**Pull request by haltcase**
* `for-in` destructuring
* Decimals requiring leading / trailing numbers
* Standard library ( lodash + inline )
* Bitwise operator removals ( in favor of standard library functions )
* Object loops now use `Object.keys()` instead of `hasOwnProperty()` checks

### CODE REVIEW

File: pages/docs/docs.md [hunk 1]
```diff
@@ -669,11 +669,9 @@ Only values:
     for val v in obj:
       print(v)
 
-Note the `hasOwnProperty` check.
-
-This implementation is subject to change in the future,
-most likely by iterating over `Object.keys(obj)` with a `for-;;`
-to eliminate the `hasOwnProperty` check, which incurs a performance hit.
+Note the use of `Object.keys()` under the hood, as this will only iterate over
+_own_ keys, not _inherited_ ones. Use a [Traditional `for-in`](#traditional-for-in)
```
Comment (rattrayalex) [inline pages/docs/docs.md]: not a big deal at all, but I would non-italicize "inherited" and lowercase "traditional". Not worth doing if it's the only thing that'll require a push

File: pages/docs/docs.md [hunk 2]
```diff
@@ -695,9 +693,28 @@ for performance, `for-;;` is recommended:
...
+
+The full power of [destructuring syntax](https://mdn.io/destructuring_assignment) is possible here:
+
+    for elem { props: { color, size: [w, h] }  } in arr:
+      print(color, w, h)
+
 ### Traditional `for-in`
 
-If you wish to iterate over all keys of an object without a `hasOwnProperty` check,
+If you wish to iterate over all owned _and inherited_ keys of an object,
```
Comment (rattrayalex) [inline pages/docs/docs.md:717]: 👍

File: pages/docs/docs.md [hunk 3]
```diff
@@ -1174,7 +1199,7 @@ In LightScript, accessing an index or property using `[]` requires an indent:
 
 The required indent is relative to the line that starts a subscript chain.
 
-Note that this rule also applies to the "numberical index access" feature:
+Note that this rule also applies to the "numerical index access" feature:
```
Comment (rattrayalex) [inline pages/docs/docs.md:1202]: 😅

File: pages/docs/docs.md [hunk 4]
```diff
@@ -1271,16 +1296,15 @@ but are grouped here for convenience.
...
 
-### `~`
+### Bitwise operators
 
-The unary Bitwise NOT `~` is not included in the language, as it has been repurposed
+All bitwise operators have been removed. The unary Bitwise NOT `~` has been repurposed
 for [Tilde Calls](#tilde-calls).
 
-Instead, you may use the `bitwiseNot()` function provided by the [standard library](#standard-library).
+Instead you may use the replacements provided by the [standard library](#standard-library).
```
Comment (rattrayalex) [inline pages/docs/docs.md]: might be nice to add "(eg; `bitwiseNot(x)`)" or equiv

File: pages/docs/docs.md [hunk 5]
```diff
@@ -1324,6 +1348,21 @@ or use parens like so:
     ):
       doStuff()
 
+### Decimals
+
+Numbers in LightScript must begin and end with a number character. That means they
```
Comment (rattrayalex) [inline pages/docs/docs.md]: technically, this isn't true as octals and such are still allowed

Comment (rattrayalex) [reply pages/docs/docs.md]: Also technically, they're floats not decimals, though I think that distinction isn't really useful here (I'm just nervous about getting people's hopes up about an actual Decimal type)

**Review by rattrayalex — APPROVED** [review]
Awesome, thanks so much for doing this! Happy to ship as-is, though feel free to make a few of the mentioned changes (none are impt)

File: pages/docs/docs.md (hunk 5 above)
Comment (haltcase) [reply pages/docs/docs.md]: I really waffled on all the wording in this section. Will try to improve.

Comment (haltcase) [reply pages/docs/docs.md]: `Float-style numbers` ?

Comment (rattrayalex) [reply pages/docs/docs.md]: Yeah, me too. I call them "Numbers with decimals". I might reword the section as:

```
### Numbers with decimals

Numbers in LightScript cannot begin or end with a `.`:

**Comment by rattrayalex** [comment]
Awesome! I'll merge in the morning alongside a deploy

**Pull request merged**